### ISSUE TITLE
Ruby Segmentation Fault

### ISSUE THREAD

**Issue by blowmage** [google/protobuf#3667]
Last night the google-cloud-ruby project had a segfault on the Travis-CI build system used to run tests on Mac OS X. This segfault was using the latest google-protobuf 3.4.1.1 (universal-darwin). The segfault looks to be happening as the generated ruby protobug code is being loaded, and not during an active call.

System details:
 
* gRPC language and version: grpc 1.6.0 (universal-darwin)
* Protobuf version: google-protobuf 3.4.1.1 (universal-darwin)
* Operating System: MacOS Sierra
* Runtime version: ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin16]

<details><summary>segfault details</summary>

```
/Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/v1/cloud_speech_pb.rb:85: [BUG] Segmentation fault at 0x00000000000000
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin16]

-- Crash Report log information --------------------------------------------
   See Crash Report log file under the one of following:
     * ~/Library/Logs/CrashReporter
     * /Library/Logs/CrashReporter
     * ~/Library/Logs/DiagnosticReports
     * /Library/Logs/DiagnosticReports
   for more details.
Don't forget to include the above Crash Report log file in bug reports.

-- Control frame information -----------------------------------------------
c:0047 p:---- s:0199 e:000198 CFUNC  :add_message
c:0046 p:0151 s:0194 E:000a70 BLOCK  /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/v1/cloud_speech_pb.rb:85 [FINISH]
c:0045 p:---- s:0191 e:000190 CFUNC  :instance_eval
c:0044 p:---- s:0188 e:000187 CFUNC  :build
c:0043 p:0083 s:0184 E:0022d0 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/v1/cloud_speech_pb.rb:12 [FINISH]
c:0042 p:---- s:0181 e:000180 CFUNC  :require
c:0041 p:0053 s:0176 e:000175 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/v1/speech_client.rb:32 [FINISH]
c:0040 p:---- s:0173 e:000172 CFUNC  :require
c:0039 p:0008 s:0168 e:000167 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/v1.rb:14 [FINISH]
c:0038 p:---- s:0165 e:000164 CFUNC  :require
c:0037 p:0017 s:0160 e:000159 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/credentials.rb:17 [FINISH]
c:0036 p:---- s:0157 e:000156 CFUNC  :require
c:0035 p:0017 s:0152 e:000151 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/service.rb:17 [FINISH]
c:0034 p:---- s:0149 e:000148 CFUNC  :require
c:0033 p:0026 s:0144 e:000143 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech/project.rb:18 [FINISH]
c:0032 p:---- s:0141 e:000140 CFUNC  :require
c:0031 p:0017 s:0136 e:000135 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/lib/google/cloud/speech.rb:17 [FINISH]
c:0030 p:---- s:0133 e:000132 CFUNC  :require
c:0029 p:0071 s:0128 e:000127 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/test/helper.rb:22 [FINISH]
c:0028 p:---- s:0125 e:000124 CFUNC  :require
c:0027 p:0008 s:0120 e:000119 TOP    /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/test/google/cloud/speech/audio/process_test.rb:15 [FINISH]
c:0026 p:---- s:0117 e:000116 CFUNC  :require_relative
c:0025 p:0010 s:0112 e:000111 BLOCK  /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/Rakefile:10 [FINISH]
c:0024 p:---- s:0108 e:000107 CFUNC  :each
c:0023 p:0032 s:0104 e:000103 BLOCK  /Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/Rakefile:10 [FINISH]
c:0022 p:0030 s:0101 e:000100 BLOCK  /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:248 [FINISH]
c:0021 p:---- s:0097 e:000096 CFUNC  :each
c:0020 p:0127 s:0093 e:000092 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:243
c:0019 p:0085 s:0088 e:000087 BLOCK  /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:187
c:0018 p:0015 s:0085 e:000084 METHOD /Users/travis/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/monitor.rb:214
c:0017 p:0028 s:0081 e:000080 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:180
c:0016 p:0039 s:0073 e:000072 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:173
c:0015 p:0038 s:0067 e:000066 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:152
c:0014 p:0010 s:0059 e:000058 BLOCK  /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:108 [FINISH]
c:0013 p:---- s:0055 e:000054 CFUNC  :each
c:0012 p:0050 s:0051 e:000050 BLOCK  /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:108
c:0011 p:0030 s:0048 e:000047 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:117
c:0010 p:0009 s:0043 e:000042 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:102
c:0009 p:0022 s:0039 e:000038 BLOCK  /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:80
c:0008 p:0006 s:0036 e:000035 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:178
c:0007 p:0009 s:0031 e:000030 METHOD /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:77
c:0006 p:0024 s:0027 e:000026 TOP    /Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/exe/rake:27 [FINISH]
c:0005 p:---- s:0024 e:000023 CFUNC  :load
c:0004 p:0152 s:0019 E:0019a0 EVAL   /Users/travis/.rvm/gems/ruby-2.4.0/bin/rake:23 [FINISH]
c:0003 p:---- s:0014 e:000013 CFUNC  :eval
c:0002 p:0129 s:0007 E:000468 EVAL   /Users/travis/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15 [FINISH]
c:0001 p:0000 s:0003 E:0006a0 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
/Users/travis/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `<main>'
/Users/travis/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `eval'
/Users/travis/.rvm/gems/ruby-2.4.0/bin/rake:23:in `<main>'
/Users/travis/.rvm/gems/ruby-2.4.0/bin/rake:23:in `load'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/exe/rake:27:in `<top (required)>'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:77:in `run'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:80:in `block in run'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:102:in `top_level'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:117:in `run_with_threads'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:108:in `block in top_level'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:108:in `each'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:108:in `block (2 levels) in top_level'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/application.rb:152:in `invoke_task'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:173:in `invoke'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:180:in `invoke_with_call_chain'
/Users/travis/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:187:in `block in invoke_with_call_chain'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:243:in `execute'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:243:in `each'
/Users/travis/.rvm/gems/ruby-2.4.0/gems/rake-11.3.0/lib/rake/task.rb:248:in `block in execute'
/Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/Rakefile:10:in `block in <top (required)>'
/Users/travis/build/[secure]/google-cloud-ruby/google-cloud-speech/Rakefil/Users/travis/.travis/job_stages: line 166: shell_session_update: command not found
```

</details>

**Comment by TeBoring** [comment]
Dup of #3649

**Comment by TeBoring** [comment]
The problem happens in add_message, which is very likely to have been fixed by #3708.
